apis/core/lib/domain/csv-mapping/update.ts

```diff
@@ -0,0 +1,58 @@
+import { GraphPointer } from 'clownface'
+import { NamedNode } from 'rdf-js'
+import { CsvSource } from '@cube-creator/model'
+import RdfResource from '@tpluscode/rdfine'
+import { schema } from '@tpluscode/rdf-ns-builders'
+import { ResourceStore } from '../../ResourceStore'
+import { resourceStore } from '../resources'
+import { NotFoundError } from '../../errors'
+import { error } from '../../log'
+import { Readable } from 'stream'
+import * as s3 from '../../storage/s3'
+import { loadFileHeadString } from '../csv/file-head'
+import { parse } from '../csv'
+import { sampleValues } from '../csv/sample-values'
+
+interface UpdateCsvSourceCommand {
+  resource: GraphPointer<NamedNode>
+  store?: ResourceStore
+  fileStorage?: s3.FileStorage
+}
+
+export async function update({
+  resource,
+  store = resourceStore(),
+  fileStorage = s3,
+}: UpdateCsvSourceCommand): Promise<GraphPointer> {
+  const changed = RdfResource.factory.createEntity<CsvSource>(resource)
+  const csvSource = await store.getResource<CsvSource>(resource.term)
+  if (!csvSource) {
+    throw new NotFoundError(resource)
+  }
+
+  csvSource.name = changed.name
```

Comment:
Now that you removed the name from the shape, we'd need to remove this assignment too, lest a request clears the name

Or we could bring back the property shape with `dash:hidden`

Reply:
Ok I'll re-add the property with `dash:hidden`